tools/compress/compression_converter.cpp

```diff
@@ -72,190 +72,67 @@ bool CompressionConverter::WriteFileHeader(const format::FileHeader&
     return FileTransformer::WriteFileHeader(header, output_options);
 }
 
-bool CompressionConverter::ProcessFunctionCall(const format::BlockHeader& block_header, format::ApiCallId call_id)
+bool CompressionConverter::ProcessFunctionCall(decode::ParsedBlock& parsed_block, const decode::FunctionCallArgs& args)
 {
-    size_t           parameter_buffer_size = static_cast<size_t>(block_header.size) - sizeof(call_id);
-    uint64_t         uncompressed_size     = 0;
-    format::ThreadId thread_id             = 0;
-
-    bool success = ReadBytes(&thread_id, sizeof(thread_id));
+    parsed_block.Decompress(GetBlockParser());
+    bool success = parsed_block.IsReady();
 
     if (success)
     {
-        parameter_buffer_size -= sizeof(thread_id);
-
-        if (format::IsBlockCompressed(block_header.type))
-        {
-            success = ReadBytes(&uncompressed_size, sizeof(uncompressed_size));
-
-            if (success)
-            {
-                parameter_buffer_size -= sizeof(uncompressed_size);
-
-                GFXRECON_CHECK_CONVERSION_DATA_LOSS(size_t, uncompressed_size);
```

Comment:
AFAICT, you're dropping a bunch of these.  I'm not sure how useful they were in the past, but wondering if you did this intentionally?